functions/github/webhooks.js

```diff
@@ -0,0 +1,154 @@
+const {
+  getReleaseIdFromBranch,
+  updateCheckRunStatus,
+} = require("../database/database.js");
+const {
+  syncReleaseState,
+} = require("../handlers/handlers.js");
+const {
+  verifySignature,
+} = require("./github.js");
+const {Octokit} = require("@octokit/rest");
+const {error, log} = require("firebase-functions/logger");
+const {defineSecret} = require("firebase-functions/params");
+const GITHUB_WEBHOOK_SECRET = defineSecret("GITHUB_WEBHOOK_SECRET");
+const GITHUB_TOKEN = defineSecret("GITHUB_TOKEN");
+
+/**
+  * Handles a GitHub check run event. This function is called when a check run
+  * is created, completed, or rerequested. It updates the release state if
+  * necessary.
+  *
+  * @param {Object} req - The request object.
+  * @param {Object} res - The response object.
+  * @return {Promise} A promise that resolves when the request is complete.
+  */
+async function githubWebhook(req, res) {
+  if (!verifySignature(req, GITHUB_WEBHOOK_SECRET.value())) {
+    return res.status(401).send("Unauthorized");
+  }
+
+  const payload = req.body;
+  const eventType = req.headers["x-github-event"];
+
+  log("Received GitHub webhook", {payload: payload, eventType: eventType});
+
+  try {
+    if (eventType === "check_run") {
+      await handleCheckRunEvent(payload);
+    } else if (eventType === "pull_request") {
+      await handlePullRequestEvent(payload);
+    }
```

Comment:
Is this if statement exhaustive?

Reply:
It could be, depending on which events we choose to send when configuring the webhooks. 

If `eventType` is not an event that we care about, then we just return 200, the only other thing we might want to do is log the event for debugging purposes. But, if we were to add logs for other `eventType` values, they would just add noise to the logs.